### ISSUE TITLE
Expo Go v2.22.2 crashes loading any Snack with Android test phone

### ISSUE THREAD

**Issue by Codelica** [expo/expo#14914]
### Summary

One of our Android test phones is a Samsung Galaxy A5, which has been working fine with Expo Go local development and running Snacks prior to 2.22.2.  With 2.22.2, development against a local Expo dev server still works fine, but attempting to load literally any Snack results in a modal with "Expo Go has stopped. Open app again... Send feedback"

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

Snacks on 42, 41, etc.  Doesn't seem to matter.

### Environment

Isn't related to local dev service.  

### Reproducible demo or steps to reproduce from a blank project

Basically bring up any Snack on this Galaxy A5 :)  I've tried using the bare, default Snack saved as a reference.   I've also tried completely un-installing Expo Go and re-installing it after reboot several times.  Nothing seems to help.   Any thoughts would be appreciated!

**Comment by AmenBreak175** [comment]
Same thing is happening with my Android Galaxy A3 (2017).

**Comment by sclea** [comment]
Confirm the same behaviour.
ZTE Blade V8 mini
Android 7.0
Expo client 2.22.3, clean first install
Trying to run stock sample snack https://snack.expo.dev

Stack trace starts with mostly the same lines as of [Codelica](https://github.com/expo/expo/issues/14914#issuecomment-953998685):

![image](https://user-images.githubusercontent.com/15073709/143533800-ec8c2d6e-fe95-4101-beb2-198952a646ed.png)
![image](https://user-images.githubusercontent.com/15073709/143533783-b8aa01c4-2454-4ec0-9230-3d50693b4495.png)

**Comment by ArturoDucasse** [comment]
It's happening to me as well, also when I try to login in my account.

**Comment by gilex-dev** [comment]
Same issue on Samsung Galaxy S7 (Android 8.0.0) and client version 2.23.2 (clean install) here.

- App crashes when I try to login
- App crashes when I try to load any project from snack (via QR-Code or from browser intent)

**Comment by lubinakov** [comment]
I have the same issues with my Galaxy A5 (Android 8.0.0.) and client version 2.23.2 (newly re-installed today). The app crashes when I try to open a project.
